src/weapons.cpp

```diff
@@ -0,0 +1,1526 @@
+    return false;
+}
+
+int CurrentWeaponToItemID(struct ss_animation_s* ss_anim)
+{
+    int32_t ver = World_GetVersion();
+
+    if (GET_MODEL(TR_MODEL_PISTOL))
+    {
+        return ITEM_PISTOL;
+    }
+    else if ((ver > TR_I_UB) && GET_MODEL(TR_MODEL_SHOTGUN))
+    {
+        return ITEM_SHOTGUN;
+    }
+
+    if (ver < TR_II)  // tomb raider 1
+    {
+        if (GET_MODEL(TR1_MODEL_SHOTGUN))
+        {
+            return ITEM_SHOTGUN;
+        }
+        else if (GET_MODEL(TR1_MODEL_MAGNUM))
+        {
+            return ITEM_MAGNUMS;
+        }
+        else if (GET_MODEL(TR1_MODEL_UZI))
+        {
+            return ITEM_UZIS;
+        }
+    }
+    else if (ver < TR_III) // tomb raider 2
+    {
+        if (GET_MODEL(TR2_MODEL_AUTOMAGS))
+        {
+            return ITEM_AUTOMAGS;
+        }
+        else if (GET_MODEL(TR2_MODEL_UZI))
+        {
+            return ITEM_UZIS;
+        }
+        else if (GET_MODEL(TR2_MODEL_M16))
+        {
+            return ITEM_M16;
+        }
+        else if (GET_MODEL(TR2_MODEL_GRENADEGUN))
+        {
+            return ITEM_GRENADEGUN;
+        }
+        else if (GET_MODEL(TR2_MODEL_HARPOONGUN))
+        {
+            return ITEM_HARPOONGUN;
+        }
+    }
+    else if (ver < TR_IV) // tomb raider 3
+    {
+        if (GET_MODEL(TR3_MODEL_DESERTEAGLE))
+        {
+            return ITEM_DESERTEAGLE;
+        }
+        else if (GET_MODEL(TR3_MODEL_UZI))
+        {
+            return ITEM_UZIS;
+        }
+        else if (GET_MODEL(TR3_MODEL_MP5))
+        {
+            return ITEM_MP5;
+        }
+        else if (GET_MODEL(TR3_MODEL_ROCKETGUN))
+        {
+            return ITEM_ROCKETGUN;
+        }
+        else if (GET_MODEL(TR3_MODEL_GRENADEGUN))
+        {
+            return ITEM_GRENADEGUN;
+        }
+        else if (GET_MODEL(TR3_MODEL_HARPOONGUN))
+        {
+            return ITEM_HARPOONGUN;
+        }
+    }
+    else if (ver < TR_V)         // tomb raider 4 and 5
+    {
+        if (GET_MODEL(TR4C_MODEL_UZI))
+        {
+            return ITEM_UZIS;
+        }
+        else if (GET_MODEL(TR4C_MODEL_CROSSBOW))
+        {
+            return ITEM_CROSSBOW;
+        }
+        else if (GET_MODEL(TR4C_MODEL_GRENADEGUN))
+        {
+            return ITEM_GRENADEGUN;
+        }
+        else if (GET_MODEL(TR4C_MODEL_REVOLVER))
+        {
+            return ITEM_REVOLVER;
+        }
+    }
+    else if (ver == TR_V)
+    {
+        if (GET_MODEL(TR4C_MODEL_UZI))
+        {
+            return ITEM_UZIS;
+        }
+        else if (GET_MODEL(TR4C_MODEL_CROSSBOW))
+        {
+            return ITEM_GRAPPLEGUN;  // it's grappling gun !
+        }
+        else if (GET_MODEL(TR4C_MODEL_GRENADEGUN))
+        {
+            return ITEM_MP5;  // it's h&k gun !
+        }
+        else if (GET_MODEL(TR4C_MODEL_REVOLVER))
+        {
+            return ITEM_REVOLVER;
+        }
+    }
+
+    // no id found
+    return NULL;

```

Comment:
```suggestion
    return 0;
```
Is 0 an invalid ID (I did not dig into IDs code, it's just out of curiosity) ?

Reply:
0 is null because 1 is for pistol #fixed